src/masternode-payments.cpp

```diff
@@ -367,17 +367,22 @@ void CMasternodePayments::ProcessMessageMasternodePayments(CNode* pfrom, std::st
 
         if(pfrom->nVersion < MIN_MNW_PEER_PROTO_VERSION) return;
 
-        if(chainActive.Tip() == NULL) return;
+        int nHeight;
+        {
+            TRY_LOCK(cs_main, locked);
```

Comment:
Careful, I don't think TRY_LOCK necessarily guarantees you won't deadlock, unless any other held locks are also TRY_LOCK'd.

Still given a choice between a rare deadlock and a rare segfault I would probably go for the deadlock, it's probably less likely to be exploitable.  I recommend this be tested for a few days though before it's merged, just too make sure the deadlocks aren't too common.